ssl/ssl_sess.c

```diff
@@ -877,13 +877,7 @@ int SSL_set_session(SSL *s, SSL_SESSION *session)
     const SSL_METHOD *meth;
 
     if (session != NULL) {
-        meth = s->ctx->method->get_ssl_method(session->ssl_version);
-        if (meth == NULL)
-            meth = s->method->get_ssl_method(session->ssl_version);
-        if (meth == NULL) {
-            SSLerr(SSL_F_SSL_SET_SESSION, SSL_R_UNABLE_TO_FIND_SSL_METHOD);
-            return (0);
-        }
+        meth = s->ctx->method;
 
         if (meth != s->method) {
             if (!SSL_set_ssl_method(s, meth))
```

Comment:
I think you want to lose this (and line 903) too? The only case where it runs is if version negotiation has already happened. This API already explodes horrible if called after the `SSL` has started, so it would only be relevant if you reuse the object with `SSL_clear`. (This is a really bad feature and needs to go away. Alas, I wasn't able to for BoringSSL. Found enough consumers that did it.) But, in that case, the method reset belongs in `SSL_clear`, not in `SSL_set_session`.